Source/drlg_l2.cpp

```diff
@@ -246,93 +246,69 @@ void InitDungeon()
 
 void L2LockoutFix()
 {
-	signed int v0;   // ecx
-	char *v1;        // eax
-	signed int v2;   // edx
-	signed int v3;   // ecx
-	signed int v4;   // edi
-	signed int v5;   // eax
-	char *v6;        // esi
-	signed int v7;   // edx
-	char v8;         // al
-	unsigned int v9; // ecx
-	signed int v10;  // eax
-	char v11;        // dl
-	signed int v12;  // esi
-	char v13;        // bl
-	char *v14;       // edx
-
-	v0 = 0;
-	do {
-		v1 = (char *)dungeon + v0;
-		v2 = 40;
-		do {
-			if (*v1 == 4 && *(v1 - 40) != 3)
-				*v1 = 1;
-			if (*v1 == 5 && *(v1 - 1) != 3)
-				*v1 = 2;
-			v1 += 40;
-			--v2;
-		} while (v2);
-		++v0;
-	} while (v0 < 40);
-	v3 = 1;
-	do {
-		v4 = 1;
-		do {
-			v5 = v4;
-			if (dflags[v4][v3] >= 0) {
-				v6 = (char *)&dungeon[v5][v3];
-				if ((*v6 == 2 || *v6 == 5) && *(v6 - 1) == 3 && dungeon[v5][v3 + 1] == 3) {
-					v7 = 0;
-					while (1) {
-						v8 = *v6;
-						if (*v6 != 2 && v8 != 5)
-							break;
-						if (*(v6 - 1) != 3 || v6[1] != 3)
-							break;
-						if (v8 == 5)
-							v7 = 1;
-						++v4;
-						v6 += 40;
-					}
-					if (!v7 && dflags[v4 - 1][v3] >= 0) // dflags[-1][]
-						dungeon[v4 - 1][v3] = 5;        // dungeon[-1][]
+	int i, j;
+	BOOL doorok;
+
+	for(j = 0; j < DMAXY; j++) {
+		for(i = 0; i < DMAXX; i++) {
+			if(dungeon[i][j] == 4 && dungeon[i - 1][j] != 3) {
+				dungeon[i][j] = 1;
+			}
+			if(dungeon[i][j] == 5 && dungeon[i][j - 1] != 3) {
+				dungeon[i][j] = 2;
+			}
+		}
+	}
+	for(j = 1; j < DMAXY - 1; j++) {
+		for(i = 1; i < DMAXX - 1; i++) {
+			if(dflags[i][j] & 0x80) {

```

Comment:
```suggestion
			if(dflags[i][j] & DFLAG_EXPLORED) {
```

Reply:
Andi, would you invite me as a collaborator to your fork? That was I can merge the trivial suggestions directly.

>  You do not have permission to push to this repository.

Reply:
Sure! Invited. However, are you sure the DFLAG enum is the same for both `dFlags` and `dflags`?

Reply:
> Sure! Invited. However, are you sure the DFLAG enum is the same for both `dFlags` and `dflags`?

Oh, that's a good point. Actually I'm not sure if that's the case.